ChurchCRM/CRM
# Redirect after Login

#### What's this PR do?
When loading a URL that requires authentication from a browser session which has timed out, or has not yet been logged in, this PR allows the login process to direct the user to their intended URL after successful authentication

```

Any Page Request --> Functions.php ---(If logged in and not expired)---> Load page
                                     \--(If Not logged in, or expired)---> Redirect to Login.php?Location=<Original requested location>


Login.php --(If location is in GET request)-->store location in new Session
           \--(if POST, and location is in previous session) --> Redirect to location
             \---(if POST and no location)---> redirect to Menu.php

```

#### Screenshots (if appropriate)

#### What Issues does it Close?

Closes #377

#### What are the relevant tickets?

#### Any background context you want to provide?

#### Where should the reviewer start?
Review the conversation in ticket #377 
Review the 3 files changed.

#### How should this be manually tested?

*Primary Use Case*
1) Begin from a logged-in session.  
2) Navigate to a page of interest (person, family, or group view page.)
3) Copy the URL
4) Log out 
5) Navigate to the previously copied URL
6)  You should see a login screen
7) Authenticate successfully
8) You should see the page at the copied URL

*Smoke Tests*
1) Follow all steps above, but try using invalid URLs. 
    a)  Also try manipulating the URL in the GET requests and re-loading login a number of times with varying GET parameters.  These should all yield the following result: 
2) You should be redirected to the main menu page.


#### How should the automated tests treat this?

#### Questions:
- [ ] Is there a related website / article to substantiate / explain this change?
- [ ] Does the development wiki need an update?
- [ ] Does the user documentation wiki need an update?
- [ ] Does this need to add new data to the demo database
